drools/src/test/java/com/datatorrent/drools/rules/DroolListenerTest/RulesFiredListenerTest.java

```diff
@@ -0,0 +1,137 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.drools.rules.DroolListenerTest;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.kie.api.KieServices;
+import org.kie.api.definition.rule.Rule;
+import org.kie.api.event.rule.AgendaEventListener;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.StatelessKieSession;
+
+import com.datatorrent.drools.rules.DroolListener.RulesFiredListener;
+import com.datatorrent.drools.rules.Product;
+
+public class RulesFiredListenerTest
+{
+
+  private StatelessKieSession kieSession;
+  Map<Object,List<Rule>> factRule;
+  Map<Rule,Integer> ruleCount;
+  protected void initialize()
+  {
+    KieServices kieServices = KieServices.Factory.get();
+    KieContainer kieContainer;
+    kieContainer = kieServices.getKieClasspathContainer();
+    kieSession = kieContainer.newStatelessKieSession("ksession-rule");
+    factRule = new HashMap<>();
+    ruleCount = new HashMap<>();
+    AgendaEventListener listener=new RulesFiredListener(factRule, ruleCount);
+    kieSession.addEventListener(listener);
+  }
+
+  protected List<Object> getListForRuleMatch()
+  {
+    Product gold1 = new Product(1,"gold",0);
+    Product gold2 = new Product(2,"gold",0);
+    Product gold3 = new Product(3,"gold",0);
+    Product gold4 = new Product(4,"gold",0);
+
+    Product diamond1 = new Product(3,"diamond",0);
+    Product diamond2 = new Product(3,"diamond",0);
+    Product diamond3 = new Product(3,"diamond",0);
+
+    List<Object> toModify = new ArrayList<>();
+    toModify.add(gold1);
+    toModify.add(gold2);
+    toModify.add(gold3);
+    toModify.add(gold4);
+    toModify.add(diamond1);
+    toModify.add(diamond2);
+    toModify.add(diamond3);
+
+    return toModify;
+  }
+
+  protected List<Object> getListForNoRulesMatch()
+  {
+    Product gold1 = new Product(1,"fakegold",0);
+    Product gold2 = new Product(21,"fakegold",0);
+    Product gold3 = new Product(3,"fakegold",0);
+    Product gold4 = new Product(4,"fakegold",0);
+
+    Product diamond1 = new Product(3,"fakediamond",0);
+    Product diamond2 = new Product(3,"fakediamond",0);
+    Product diamond3 = new Product(3,"fakediamond",0);
+
+    List<Object> toModify = new ArrayList<>();
+    toModify.add(gold1);
+    toModify.add(gold2);
+    toModify.add(gold3);
+    toModify.add(gold4);
+    toModify.add(diamond1);
+    toModify.add(diamond2);
+    toModify.add(diamond3);
+
+    return toModify;
+  }
+
+  @Test
+  public void RulesFiredListenerTestFactRule() throws Exception {
```

Comment:
function name should start with lowercase, also we normally name test fuctions as testxxxx